pkg/controller/backupbucket/reconciler.go

```diff
@@ -0,0 +1,179 @@
+	if err := r.actuator.Reconcile(ctx, bb); err != nil {
+		msg := "Error reconciling backupbucket"
+		_ = r.updateStatusError(ctx, extensionscontroller.ReconcileErrCauseOrErr(err), bb, operationType, msg)
+		r.logger.Error(err, msg, "backupbucket", bb.Name)
+		return extensionscontroller.ReconcileErr(err)
+	}
+
+	msg := "Successfully reconciled backupbucket"
+	r.logger.Info(msg, "backupbucket", bb.Name)
+	r.recorder.Event(bb, corev1.EventTypeNormal, EventBackupBucketReconciliation, msg)
+	if err := r.updateStatusSuccess(ctx, bb, operationType, msg); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) delete(ctx context.Context, bb *extensionsv1alpha1.BackupBucket) (reconcile.Result, error) {
+	hasFinalizer, err := extensionscontroller.HasFinalizer(bb, FinalizerName)
+	if err != nil {
+		r.logger.Error(err, "Could not instantiate finalizer deletion")
+		return reconcile.Result{}, err
+	}
+	if !hasFinalizer {
+		r.logger.Info("Deleting backupbucket causes a no-op as there is no finalizer.", "backupbucket", bb.Name)
+		return reconcile.Result{}, nil
+	}
+
+	operationType := gardencorev1alpha1helper.ComputeOperationType(bb.ObjectMeta, bb.Status.LastOperation)
+	if err := r.updateStatusProcessing(ctx, bb, operationType, "Deleting the backupbucket"); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	r.logger.Info("Starting the deletion of backupbucket", "backupbucket", bb.Name)
+	r.recorder.Event(bb, corev1.EventTypeNormal, EventBackupBucketDeletion, "Deleting the backupbucket")
+	if err := r.actuator.Delete(r.ctx, bb); err != nil {
+		msg := "Error deleting backupbucket"
+		r.recorder.Eventf(bb, corev1.EventTypeWarning, EventBackupBucketDeletion, "%s: %+v", msg, err)
+		_ = r.updateStatusError(ctx, extensionscontroller.ReconcileErrCauseOrErr(err), bb, operationType, msg)
+		r.logger.Error(err, msg, "backupbucket", bb.Name)
+		return extensionscontroller.ReconcileErr(err)
+	}
+
+	msg := "Successfully deleted backupbucket"
+	r.logger.Info(msg, "backupbucket", bb.Name)
+	r.recorder.Event(bb, corev1.EventTypeNormal, EventBackupBucketDeletion, msg)
+	if err := r.updateStatusSuccess(ctx, bb, operationType, msg); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	r.logger.Info("Removing finalizer.", "backupbucket", bb.Name)
+	if err := extensionscontroller.DeleteFinalizer(ctx, r.client, FinalizerName, bb); err != nil {
+		r.logger.Error(err, "Error removing finalizer from backupbucket", "backupbucket", bb.Name)
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) updateStatusProcessing(ctx context.Context, bb *extensionsv1alpha1.BackupBucket, lastOperationType gardencorev1alpha1.LastOperationType, description string) error {
+	bb.Status.LastOperation = extensionscontroller.LastOperation(lastOperationType, gardencorev1alpha1.LastOperationStateProcessing, 1, description)
+	return r.client.Status().Update(ctx, bb)
```

Comment:
Use the `TryUpdateStatus` functions like in the other controllers.

Reply:
I referred (this)[https://github.com/gardener/gardener-extensions/blob/master/pkg/controller/controlplane/reconciler.go#L169] Anyways, if that's the right way then i'll change it at both places.

Reply:
Hm, ok, it should be changed in the control plane controller as well (cc @stoyanr). Let's do that separately so that you can focus on the backup stuff.